manager/controlapi/secret.go

```diff
@@ -12,13 +17,40 @@ import (
 // MaxSecretSize is the maximum byte length of the `Secret.Spec.Data` field.
 const MaxSecretSize = 500 * 1024 // 500KB
 
+var isValidSecretName = regexp.MustCompile(`^[a-zA-Z0-9](?:[a-zA-Z0-9-_.]{0,62}[a-zA-Z0-9])*$`)
+
+// assumes spec is not nil
+func secretFromSecretSpec(spec *api.SecretSpec) *api.Secret {
+	return &api.Secret{
+		ID:         identity.NewID(),
+		Spec:       *spec,
+		SecretSize: uint32(len(spec.Data)),
+		Digest:     digest.FromBytes(spec.Data).String(),
```

Comment:
No, don't create a new protobuf type. Just tell the compiler to generate this field with type `digest.Digest` using the custom type extension.